### ISSUE TITLE
Surface _notification

### ISSUE THREAD

**Issue by migueldeicaza** [migueldeicaza/SwiftGodot#2]
Surface _notification, we are getting the event, they are just not being posted to the object yet.

Need to decide if we create a big "Notification" enum with all possible notifications, or if "Notification" should be narrowed to the type they are declared on, and whether it is an enum or a constant.   In the former case, we can use an enum, in the latter case we would need to use constants, in case new subclasses introduce new notifications.

**Comment by LemmaEOF** [comment]
I think constants narrowed to type would be better because not all object types receive the same notifications. The only ones on base `Object` are `POSTINITIALIZE` and `PREDELETE`, then nodes and such add a bunch more. People might want to make their own custom notifications, too.

**Comment by migueldeicaza** [comment]
Update: we already surfaced all the constants in each type, so they are already there, we could reuse those to prototype at least.

I think that what needs to be done is to add a new virtual

```
open func notification(...)
```

Method in the generated Object, and then we would just call it from the notification callback in Wrapper.

**Comment by Dracks** [comment]
Hey @migueldeicaza what is the status of this notification? 

Thanks!

**Comment by migueldeicaza** [comment]
Sorry, have not had time to look at this.

**Comment by migueldeicaza** [comment]
Untested, hacked this up from a meeting from my iPad over ssh, let me know if this works:

https://gist.github.com/migueldeicaza/3dbb69f6b00e166eee414b581664bc40

**Comment by Dracks** [comment]
> Sorry, have not had time to look at this.

Hey, sorry if you missunderstand me, It's not so urgent, technically I discovered yesterday about the notification, I'm doing a tutorial of godot (and I'm transforming it into Swift), thats where I found. 

Maybe this afternoon I will get time, but I don't expect a lot of time until next week.  

I only wished to raise the topic, also I found all the constants, which I was expecting to put it here. 

Thanks!
